vclib-openid/src/commonMain/kotlin/at/asitplus/wallet/lib/oidc/OidcSiopWallet.kt

```diff
@@ -113,7 +113,7 @@ class OidcSiopWallet(
         val params = kotlin.runCatching {
             Url(it).parameters.flattenEntries().toMap().decodeFromUrlQuery<AuthenticationRequestParameters>()
         }.getOrNull()
-            ?: return KmmResult.failure(OAuth2Exception(Errors.INVALID_REQUEST))
+            ?: return KmmResult.failure<AuthenticationResponseResult>(OAuth2Exception(Errors.INVALID_REQUEST))
```

Comment:
Why do we need the type parameter here, and not in other functions?

Reply:
The new KmmResult requires type declaration also for the failure case since the <Nothing>-bounded subclass is gone. Automatic inference fails here. In this particular context I'd need to really think through why the declared return type of the function is not enough

Reply:
This will probably not be necessary anymore with K2 (i.e. kotlin 2.0).
The frontend simply fails to infer the type and there is nothing we can do. resolved?